src/hooks/useFormValidation.ts

```
@@ -0,0 +1,183 @@
+
+    Object.keys(values).forEach(name => {
+      const error = validateField(name, values[name]);
+      if (error) {
+        newErrors[name] = error;
+        isFormValid = false;
+      }
+    });
+
+    setErrors(newErrors);
+    setTouched(Object.keys(values).reduce((acc, key) => ({
+      ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)